addons/binding/org.openhab.binding.unifi/src/main/java/org/openhab/binding/unifi/handler/UniFiClientHandler.java

```diff
@@ -0,0 +1,215 @@
+
+        Bridge bridge = getBridge();
+
+        if (bridge == null || bridge.getHandler() == null || !(bridge.getHandler() instanceof UniFiControllerHandler)) {
+            updateStatus(OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "You must choose a UniFi Controller for this UniFi Wireless Device.");
+            return;
+        }
+
+        if (bridge.getStatus() == OFFLINE) {
+            updateStatus(OFFLINE, BRIDGE_OFFLINE, "The UniFi Controller is currently offline.");
+            return;
+        }
+
+        // mgb: only refreshes if we we're ONLINE
+        refresh();
+
+        updateStatus(ONLINE);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Handling command = {} for channel = {}", command, channelUID);
+        if (command == REFRESH) {
+            refresh(channelUID);
+        } else {
+            logger.debug("Ignoring unsupported command = {} for channel = {} - the UniFi binding is read-only!",
+                    command, channelUID);
+        }
+    }
+
+    public void refresh() {
+        // mgb: only refresh if we're ONLINE
+        if (getThing().getStatus() == ONLINE) {
+            for (Channel channel : getThing().getChannels()) {
+                ChannelUID channelUID = channel.getUID();
+                refresh(channelUID);
+            }
+        }
+    }
+
+    private State getDefaultState(String channelID, boolean clientOnline) {
+        State state = null;
+        switch (channelID) {
+            case CHANNEL_ONLINE:
+            case CHANNEL_SITE:
+            case CHANNEL_AP:
+            case CHANNEL_ESSID:
+            case CHANNEL_RSSI:
+                state = (clientOnline ? null : UnDefType.UNDEF); // skip the update if the client is online
+                break;
+            case CHANNEL_UPTIME:
+                // mgb: uptime should default to 0 seconds
+                state = (clientOnline ? null : new DecimalType(0)); // skip the update if the client is online
+                break;
+            case CHANNEL_LAST_SEEN:
+                // mgb: lastSeen should keep the last state no matter what
+                state = null;
+                break;
+        }
+        return state;
+    }
+
+    private void refresh(ChannelUID channelUID) {
+        // mgb: only refresh if we're ONLINE
+        if (getThing().getStatus() == ONLINE) {
+            logger.debug("Refreshing channel = {}", channelUID);
+
+            UniFiController controller = ((UniFiControllerHandler) getBridge().getHandler()).getController();
+            UniFiWirelessClient client = controller.getWirelessClient(config.getMac(), config.getSite());
+            if (client == null) {
+                logger.debug("Could not find a matching client: mac = {}, site = {}", config.getMac(),
+                        config.getSite());
+            }
+            boolean clientOnline = controller.isWirelessClientOnline(client);
```

Comment:
See my comments below.

I did it like this for all the channel updates to be simple one-liner updates. All the checks around the `state = ...` lines are because there was one user where his controller returned an "unknown" client (it was missing the `is_wired` attribute in the json). This resulted in `null` devices and sites. 

I figured it was better practice to defensively guard against this possibily happening to other users in the future.